CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/utilities/MarkdownTree.java

```diff
@@ -0,0 +1,560 @@
+/*
+ * Copyright 2010-2022 Australian Signals Directorate
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.views.notes.utilities;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javafx.geometry.Insets;
+import javafx.scene.layout.Border;
+import javafx.scene.paint.Color;
+import javafx.scene.text.FontPosture;
+import javafx.scene.text.FontWeight;
+import javafx.scene.text.TextAlignment;
+import javafx.scene.text.TextFlow;
+
+/**
+ * This class makes a tree containing nodes that represent text and their
+ * respective formatting
+ *
+ * @author altair1673
+ */
+public class MarkdownTree {
```

Comment:
if you could add some unit tests for this class that would be great, keep adding finals where you can as well (there's a few missing in this class)